jetty-http2/http2-server/src/main/java/org/eclipse/jetty/http2/server/HttpChannelOverHTTP2.java

```
@@ -224,26 +237,16 @@ protected void commit(MetaData.Response info)
     @Override
     public Runnable onData(DataFrame frame, Callback callback)
     {
-        return onRequestContent(frame, callback);
-    }
-
-    public Runnable onRequestContent(DataFrame frame, final Callback callback)
-    {
-        Stream stream = getStream();
-        if (stream.isReset())
-        {
-            // Consume previously queued content to
-            // enlarge the session flow control window.
-            consumeInput();
-            // Consume immediately this content.
-            callback.succeeded();
-            return null;
-        }
-
```

Why this logic has been removed?

---

Interesting... it actually was part of @gregw 's patch that transformed the `ContentDemander` to make it atomic. I'll investigate...

---

I could not see a reason why, so I've restored it. Maybe @gregw could comment on this?

---

The call to `stream.isReset()` is a race, as the stream could be reset the nano second after it is called or at any time during the consumption of the frame that has been passed.
@sbordet is there any reason to fail fast on reset ?

If not, then not handling reset hear is the best way to test if we correctly handle a reset if the race is lost.

Besides, this check is only closing the race window a tiny bit... how often will a stream be reset in the time between a data frame is received and handed to this method?

I think the logic should be removed